vivarium/simulator/states.py

```diff
@@ -0,0 +1,316 @@
+# Plus we store different attributes of entities (e.g Agents positions and sensors) in different dataclasses
+def init_nve_state(
+        simulator_state: SimulatorState,
+        diameter: float = 5.,
+        friction: float = 0.1,
+        mass_center: float = 1.,
+        mass_orientation: float = 0.125,
+        agents_positions: Optional[Union[List[float], None]] = None,
+        objects_positions: Optional[Union[List[float], None]] = None,
+        existing_agents: Optional[Union[int, List[float], None]] = None,
+        existing_objects: Optional[Union[int, List[float], None]] = None,
+        seed: int = 0,
+        ) -> NVEState:
+    """
+    Initialize nve state with given parameters
+    """
+    n_agents = simulator_state.n_agents[0]
+    n_objects = simulator_state.n_objects[0]
+    n_entities = n_agents + n_objects
+
+    key = random.PRNGKey(seed)
+    key_pos, key_or = random.split(key)
+    key_ag, key_obj = random.split(key_pos)
+
+    # If we have a list of agents or objects positions, transform it into a jax array, else initialize random positions
+    agents_positions = _init_positions(key_ag, agents_positions, n_agents, simulator_state.box_size)
+    objects_positions = _init_positions(key_obj, objects_positions, n_objects, simulator_state.box_size)
+    # Assign their positions to each entities
+    positions = jnp.concatenate((agents_positions, objects_positions))
+
+    # Assign random orientations between 0 and 2*pi
+    orientations = random.uniform(key_or, (n_entities,)) * 2 * jnp.pi
+
+    agents_entities = jnp.full(n_agents, EntityType.AGENT.value)
+    object_entities = jnp.full(n_objects, EntityType.OBJECT.value)
+    entity_types = jnp.concatenate((agents_entities, object_entities), dtype=int)
+
+    existing_agents = _init_existing(existing_agents, n_agents)
+    existing_objects = _init_existing(existing_objects, n_objects)
+    exists = jnp.concatenate((existing_agents, existing_objects), dtype=int)
+
+    # TODO: Why is momentum set to none ?
+    return NVEState(
+        position=RigidBody(center=positions, orientation=orientations),
+        momentum=None,
+        force=RigidBody(center=jnp.zeros((n_entities, 2)), orientation=jnp.zeros(n_entities)),
+        mass=RigidBody(center=jnp.full((n_entities, 1), mass_center), orientation=jnp.full((n_entities), mass_orientation)),
+        entity_type=entity_types,
+        entity_idx = jnp.array(list(range(n_agents)) + list(range(n_objects))),
+        diameter=jnp.full((n_entities), diameter),
+        friction=jnp.full((n_entities), friction),
+        exists=exists
+        )
+
+
+# Could implement it as a static or class method
+def init_agent_state(
+        simulator_state: SimulatorState,
+        behavior: int = 1,
+        wheel_diameter: float = 2.,
+        speed_mul: float = 1.,
+        theta_mul: float = 1.,
+        prox_dist_max: float = 40.,
+        prox_cos_min: float = 0.,
+        color: str = "blue"
```

Comment:
To be discussed, I'm not sure it is super coherent to pass colors as strings, since we are in the simulator world here. For example, `behavior` above is defined as an int, not a string (whereas it is defined as a string in the client Configs, as for colors). 
Also, specifying colors as a RGB vector here will solve the issue I mentioned about matlplotlib above. Maybe the best solution would be to have a RGB <-> String converter in a utils module, that can be easily imported wherever we need this. For example here ; or in the run script